src/createXcmTypes/util/sortMultiAssetsAscending.spec.ts

```diff
@@ -8,208 +9,1327 @@ describe('sortMultiAssetsAscending', () => {
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [
+									{ Parachain: '2023' },
+									{ GeneralKey: '0xA73397cE0cCFdE92e7B23F3d0C462eF099E9E978' },
+								],
+							}
+						),
+					}),
+				},
+			},
 			{
 				fun: {
 					Fungible: '300000',
 				},
 				id: {
-					Concrete: {
-						interior: {
-							X2: [{ PalletInstance: '50' }, { GeneralIndex: '10' }],
-						},
-						parents: 0,
-					},
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [{ PalletInstance: '50' }, { GeneralIndex: '10' }],
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '300000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [{ PalletInstance: '50' }, { GeneralIndex: '1984' }],
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '200000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X3: [
+									{ Parachain: '2000' },
+									{ PalletInstance: '55' },
+									{ GeneralIndex: '0' },
+								],
+							}
+						),
+					}),
 				},
 			},
 			{
 				fun: {
 					Fungible: '100000',
 				},
 				id: {
-					Concrete: {
-						interior: {
-							X2: [{ PalletInstance: '50' }, { GeneralIndex: '1984' }],
-						},
-						parents: 0,
-					},
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 2,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X1: {
+									GeneralKey: '0xA73397cE0cCFdE92e7B23F3d0C462eF099E9E978',
+								},
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '300000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 2,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [{ PalletInstance: '50' }, { GeneralIndex: '10' }],
+							}
+						),
+					}),
+				},
+			},
+		];
+
+		const res = sortMultiAssetsAscending(multiAssets);
+
+		expect(res.length).toEqual(expected.length);
+		expect(res[0].id).toEqual(expected[0].id);
+		expect(res[res.length - 1].fun).toEqual(expected[expected.length - 1].fun);
+		expect(JSON.stringify(res)).toEqual(JSON.stringify(expected));
+	});
+
+	it('Should correctly sort an unsorted multiasset array based on fungible value when all other values are the same', () => {
```

Comment:
if all the other values are the same, is it not the same asset?

Reply:
the API should error out if the user input the same asset twice

Reply:
yep, same as above. Adding test cases for this covering assetId and MultiLocation dupes